src/dom.js

```diff
@@ -204,6 +204,49 @@ export function closestByTag(element, tagName) {
   });
 }
 
+/**
+ * Finds the closest element with the specified selector from this element
+ * @param {!Element} element
+ * @param {string} selector
+ * @return {?Element} closest ancestor if found.
+ */
+export function closestBySelector(element, selector) {
+  if (element.closest) {
+    return element.closest(selector);
+  }
+
+  return closest(element, el => {
+    return matches(el, selector);
```

Comment:
It is used in instrumentation.js. Happy to go with an alternate route but can't think of anything clean. There are two reasons for the callback:

1) instrumentation.js has a different matching behaviour then closestBySelector. 

2) `matches()` will return false for both "`.matches` method doesn't exist" and "doesn't match". As a result, if we act on the return value, we would do a second round of matching even when it is not needed(`.matches` exists and the element did not match the selector).